src/java.base/share/native/libcrexec/crexec.cpp

```diff
@@ -522,6 +545,165 @@ class Environment {
   }
 };
 
+static bool set_user_data(crlib_conf_t *conf, const char *name, const void *data, size_t size) {
+  if (!conf->argv()[ARGV_IMAGE_LOCATION]) {
+    fprintf(stderr, CREXEC "configure_image_location has not been called\n");
+    return false;
+  }
+  char fname[PATH_MAX];
+  if (snprintf(fname, sizeof(fname), "%s/%s", conf->argv()[ARGV_IMAGE_LOCATION], name) >= (int) sizeof(fname) - 1) {
+    fprintf(stderr, CREXEC "filename too long: %s/%s\n", conf->argv()[ARGV_IMAGE_LOCATION], name);
+    return false;
+  }
+  FILE *f = fopen(fname, "w");
+  if (f == NULL) {
```

Comment:
A nitpick, would also be great to replace in the other places in the file
```suggestion
  if (f == nullptr) {
```

Reply:
Would also be great to use `static_cast` and friends instead of C-style casts but since it can be cumbersome to replace sometimes I don't insist

Reply:
Done the `nullptr`. Personally I do not use C casts in my code at all but by some grepping it looks to me hotspot is mostly using the C costs so I just went with the flow. I do not see casts mentioned in the [style guide](https://github.com/openjdk/jdk/blob/master/doc/hotspot-style.md).